crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -470,69 +547,126 @@ impl SystemStage {
     }
 
     /// Logs execution order ambiguities between systems. System orders must be fresh.
-    fn report_ambiguities(&self, world: &World) {
+    fn report_ambiguities(&self, world: &mut World) {
+        let ambiguity_report =
+            world.get_resource_or_insert_with(ReportExecutionOrderAmbiguities::default);
+
+        if ambiguity_report.level == AmbiguityReportLevel::Off {
+            return;
+        }
+
         debug_assert!(!self.systems_modified);
         use std::fmt::Write;
         fn write_display_names_of_pairs(
             string: &mut String,
             systems: &[impl SystemContainer],
             mut ambiguities: Vec<(usize, usize, Vec<ComponentId>)>,
             world: &World,
+            output_prefix: &str,
         ) {
-            for (index_a, index_b, conflicts) in ambiguities.drain(..) {
+            use std::collections::hash_map;
+
+            let mut ambiguities_map: hash_map::HashMap<Vec<ComponentId>, Vec<usize>> =
+                hash_map::HashMap::new();
+
+            for (index_a, index_b, components) in ambiguities.drain(..) {
+                let systems = ambiguities_map.entry(components).or_default();
+                if !systems.contains(&index_a) {
+                    systems.push(index_a);
+                }
+                if !systems.contains(&index_b) {
+                    systems.push(index_b);
+                }
+            }
+
+            for (idx, (conflicts, systems_indices)) in ambiguities_map.drain().enumerate() {
+                let system_names = systems_indices
+                    .iter()
+                    .map(|index| systems[*index].name())
+                    .collect::<Vec<_>>();
+
+                let system_components = conflicts
+                    .iter()
+                    .map(|id| world.components().get_info(*id).unwrap().name())
+                    .collect::<Vec<_>>();
```

Comment:
While it makes for fewer warning messages, "all systems that have a conflict involving `T`" is not the same as "all systems that conflict on `T`". 

Reporting this way will actually make it harder to parse which systems actually conflict, since not all systems being grouped together are actually running at the same time. Grouping this way ignores system order.